x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/DatafeedManager.java

```diff
@@ -324,7 +325,7 @@ boolean isIsolated() {
         }
 
         public void stop(String source, TimeValue timeout, Exception e) {
-            if (isRelocating) {
+            if (isNodeShuttingDown) {
```

Comment:
I renamed this variable to make clearer what it's used for.  In the case of isolating a datafeed prior to enabling upgrade mode the datafeed may relocate to a different node when upgrade mode is eventually cancelled.  But as long as the node where this code is running remains up it needs to run the `finishHandler.accept(e);` line below otherwise the local task associated with the persistent task will linger and cause problems.